app/models/comment.php

```diff
@@ -20,17 +20,140 @@ public static function getComments($offset, $limit, $id)
         $db = DB::conn();
         $rows = $db->rows("SELECT * FROM comment 
                         WHERE thread_id = ? ORDER BY created ASC LIMIT {$offset}, {$limit}", array($id));
-       
+
+        foreach ($rows as $row) {
+            $favorites = array();
+            $user_favorites = $db->rows("SELECT * FROM favorites WHERE comment_id = ?", array($row['id']));
+            $total_favorites = Comment::countFavorites($row['id']);
+
+            foreach ($user_favorites as $v) {
+                $favorites[] = $v['user_id'];
+            }
+            $row['favorites'] = $favorites; 
+            $row['total_favorites'] = $total_favorites;
+            $comments[] = new Comment($row);
+        }  
+        return $comments;
+    }
+
+    public static function count($thread_id)
+    {
+        $db = DB::conn();
+        return (int) $db->value("SELECT COUNT(*) FROM comment WHERE thread_id = ?", array($thread_id));
+    }
+
+    public function edit($comment_id)
+    {
+        if (!$this->body) {
+            throw new RecordNotFoundException('Record Not Found');
```

Comment:
isn't the comment_id is the one to be checked if existing and throw RecordNotFound if not?